### ISSUE TITLE
`unawaited_futures` should consider `try {return future}` as unawaited

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/linter#5035]
**Describe the issue**
When returning a `Future` inside async functions, when wrapped in a `try` block, that future won't trigger an `unawaited` lint

**To Reproduce**

```dart
Future<void> fn() async {
  try {
    return Future.value();
  } catch (e) {
    ...
  }
}
```

**Expected behavior**

The `return future` line should emit an `unawaited_futures`. And awaiting it or marking it as unawaited should make the issue disappear.

**Additional context**

This is a common mistake. The `catch`/`finally` blocks may not trigger correctly when the returned future is unawaited.

**Comment by lrhn** [comment]
It's a bug that `return future;` does not await the future (https://github.com/dart-lang/sdk/issues/44395).
As such, the linter is technically correct, it's the implementation that is wrong.

Haven't been able to make it impossible to return a future yet (https://github.com/dart-lang/language/issues/870), which would force you to write the `await`.

**Comment by srawlins** [comment]
I think we can consider this a duplicate of https://github.com/dart-lang/linter/issues/2357, or vice versa if we want to update `unawaited_futures`.

**Comment by srawlins** [comment]
Duplicate of https://github.com/dart-lang/linter/issues/2250
